web/src/app/page.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+import { getProfile } from "@/openapi";
+import styles from "./page.module.css";
+import Link from "next/link";
+
+import { client } from "@/openapi/client.gen";
+
+client.setConfig({
+  baseUrl: "/api/v0",
+});
+
+export default function Home() {
+  const [helloMessage, setHelloMessage] = useState<string>("");
+  const [isAuthed, setIsAuthed] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    getProfile()
+      .then((res) => {
+        if (res.response.status == 200 && res.data) {
+          setHelloMessage(`Username: ${res.data.username}. Roles: ${res.data.roles.join(",")}`);
+          setIsAuthed(true);
+        } else {
+          setIsAuthed(false);
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  });
+  return (
+    <div className={styles.page}>
+      <h1 id="title--portal">Welcome to the ARC Services Portal</h1>
+      <main aria-label="ARC portal main content">
+        {isAuthed === false && (
+          <div>
+            <Link href="/oauth2/start">
+              <button className="btn--login" id="login">
+                Login with UCL SSO
+              </button>
+            </Link>
```

Comment:
```suggestion
            <button className="btn--login" id="login">
              <a href="/oauth2/start">Login with UCL SSO</a>
            </button>
```
think it's <a> for external links

Reply:
https://nextjs.org/docs/14/pages/api-reference/components/link

Reply:
`/oauth2/start` isn't a react route